core/src/services/fs/backend.rs

```diff
@@ -443,12 +443,14 @@ impl Accessor for FsBackend {
     }
 
     async fn copy(&self, from: &str, to: &str, _args: OpCopy) -> Result<RpCopy> {
-        tokio::fs::copy(
-            self.root.join(from.trim_end_matches('/')),
-            self.root.join(to.trim_end_matches('/')),
-        )
-        .await
-        .map_err(parse_io_error)?;
+        let from = self.root.join(from.trim_end_matches('/'));
+
+        // try to get the metadata of the source file to ensure it exists
```

Comment:
Why do we need to check this? Will `copy` already return `not found`?

Reply:
We should confirm the existence of the source file before we ensure target's parent path.

Reply:
It's not a big issue for me. Either way is fine. Let's merge and gather feedback from users.